common/build/eslint-plugin-fluid/src/test/example/no-unchecked-record-access/fileWithOnlyRecordAccess.ts

```diff
@@ -0,0 +1,147 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* Type Definitions */
+type StaticType = { a: string; b: string };
+type IndexSignatureType = { [key: string]: string };
+type NullableIndexSignatureType = { [key: string]: string | null };
+type UndefinableIndexSignatureType = { [key: string]: string | undefined };
+
+interface NonNullableProps {
+	definitelyString: string;
+	maybeString?: string;
+}
+
+interface NestedIndexProps {
+	nested: {
+		[key: string]: string;
+	};
+}
+
+/* Constants and Variables */
+const a = "a";
+const b = "b";
+const indexedRecordOfStrings: IndexSignatureType = { a: "hello", b: "goodbye" };
+const nullableIndexedRecord: NullableIndexSignatureType = { a: "hello", b: null };
+const undefinableIndexedRecord: UndefinableIndexSignatureType = { a: "hello", b: undefined };
+
+/*
+ * Accessing Properties
+ */
+
+/* Nullable index signature */
+nullableIndexedRecord.a; // ok: Accessing index property 'a' without requiring a particular result
+nullableIndexedRecord.a.length; // defect: Accessing length of index property 'a', but 'a' might be undefined or null
+nullableIndexedRecord.b?.length; // ok: Using optional chaining to access length safely handles 'undefined'
+
+/* Undefinable index signature */
+undefinableIndexedRecord.a; // ok: Accessing index property 'a' without requiring a particular result
+undefinableIndexedRecord.a.length; // defect: Accessing length of index property 'a', but 'a' might be undefined
+undefinableIndexedRecord.b?.length; // ok: Using optional chaining to access length safely handles 'undefined'
```

Comment:
change the `.b?.length` cases to `.a?.length`. We want to show good case alongside problem case. So either all `b` or all `a` use.